roles/ceph-docker-common/tasks/fetch_image.yml

```diff
@@ -1,10 +1,14 @@
 ---
 # Normal case - pull image from registry
 - name: "pull {{ ceph_docker_image }} image"
-  command: "docker pull {{ ceph_docker_registry}}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}"
+  command: "timeout {{ docker_pull_timeout }} docker pull {{ ceph_docker_registry}}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}"
   changed_when: false
+  register: docker_image
   when:
     - (ceph_docker_dev_image is undefined or not ceph_docker_dev_image)
+  until: docker_image.rc == 0
```

Comment:
@jtaleric sorry, just one last thing and we should be good, but as @leseb already asked [here](https://github.com/ceph/ceph-ansible/pull/2071#discussion_r145413250) can you put the `when` at the end of the task?
Thanks!